client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageOrderlyService.java

```diff
@@ -335,10 +335,12 @@ public ConsumerStatsManager getConsumerStatsManager() {
         return this.defaultMQPushConsumerImpl.getConsumerStatsManager();
     }
 
-    private int getMaxReconsumeTimes() {
-        // default reconsume times: Integer.MAX_VALUE
+    private int getMaxReconsumeTimes(MessageExt messageExt) {
+        if (messageExt.getMaxReConsumerTimes() >= 0) {
+            return messageExt.getMaxReConsumerTimes();
+        }
         if (this.defaultMQPushConsumer.getMaxReconsumeTimes() == -1) {
-            return Integer.MAX_VALUE;
+            return MixAll.DEFAULT_MAX_RECONSUME_TIMES;
```

Comment:
I think `private maxReconsumeTimes = -1` in DefaultMQPushConsumer is kind of unnecessary. Make it 16 just like DefaultMQPullConsumer would be simpler. So the const MixAll.DEFAULT_MAX_RECONSUME_TIMES is also unnecessary.

Reply:
I think so